### ISSUE TITLE
Magnific Popup Gallery Issue

### ISSUE THREAD

**Issue by cklenk** [nopSolutions/nopCommerce#3405]
nopCommerce version: 4.1

Steps to reproduce the problem: Upload several images to a product. Enable picture zoom in store settings. Go to that product and click on a picture, there is no way to scroll through the pictures either using the keyboard. No left or right icons show up either.

I see under the magnificPopup settings that gallery is enabled, however it doesn't seem to do anything. You can even see it on the default store (example: http://demo.nopcommerce.com/build-your-own-computer). It should behave like it shows here http://dimsemenov.com/plugins/magnific-popup/ (under gallery examples).

Thanks

**Comment by cklenk** [comment]
I actually modified the _ProductDetailsPictures, this is what I came up with. It seems to work. I even added in conditions for gallery vs single image. What do you think?

```
@model ProductDetailsModel
@{
	Html.AddScriptParts(ResourceLocation.Footer, "~/lib/magnific-popup/jquery.magnific-popup.min.js");
	Html.AddCssFileParts("~/lib/magnific-popup/magnific-popup.css");
}
<div class="gallery">
	<div class="picture">
		@if (Model.DefaultPictureZoomEnabled && Model.PictureModels.Count == 1)
		{
			<a href="@Model.DefaultPictureModel.FullSizeImageUrl" title="@Model.DefaultPictureModel.Title" id="main-product-img-lightbox-anchor-@Model.Id">
				<img alt="@Model.DefaultPictureModel.AlternateText" src="@Model.DefaultPictureModel.ImageUrl" title="@Model.DefaultPictureModel.Title" itemprop="image" id="main-product-img-@Model.Id" />
			</a>
			<script asp-location="Footer">
                $(document).ready(function () {
                    $('#main-product-img-lightbox-anchor-@Model.Id').magnificPopup({type: 'image'});
                });
			</script>
		}
		else
		{
			<img alt="@Model.DefaultPictureModel.AlternateText" src="@Model.DefaultPictureModel.ImageUrl" title="@Model.DefaultPictureModel.Title" itemprop="image" id="main-product-img-@Model.Id" />
		}
	</div>
	<span class="picture-caption">@Model.DefaultPictureModel.Title</span>
	@if (Model.PictureModels.Count > 1 && Model.DefaultPictureZoomEnabled)
	{
		<div class="picture-thumbs">
			@foreach (var picture in Model.PictureModels)
			{
				<a class="thumb-item" href="@picture.FullSizeImageUrl" alt="@picture.AlternateText" title="@picture.Title">
					<img src="@picture.ThumbImageUrl" alt="@picture.AlternateText" title="@picture.Title" data-defaultsize="@picture.ImageUrl" data-fullsize="@picture.FullSizeImageUrl" />
				</a>
			}
		</div>
		<script asp-location="Footer">
                $(document).ready(function () {
					$('.picture-thumbs').magnificPopup(
                        {
							type: 'image',
							delegate: 'a',
                            removalDelay: 300,
                            gallery: {
								enabled: true,
								navigateByImgClick: true,
								preload: [0, 1],
                                tPrev: '@T("Media.MagnificPopup.Previous")',
                                tNext: '@T("Media.MagnificPopup.Next")',
                                tCounter: '@T("Media.MagnificPopup.Counter")'
                            },
                            tClose: '@T("Media.MagnificPopup.Close")',
                            tLoading: '@T("Media.MagnificPopup.Loading")'
                        });
                });
		</script>
		<script asp-location="Footer">
            $(document).ready(function () {
                $('.thumb-item > img').on('click', function () {
                    $('#main-product-img-@Model.Id').attr('src', $(this).attr('data-defaultsize'));
                    $('#main-product-img-@Model.Id').attr('title', $(this).attr('title'));
                    $('#main-product-img-@Model.Id').attr('alt', $(this).attr('alt'));
                    $('#main-product-img-lightbox-anchor-@Model.Id').attr('href', $(this).attr('data-fullsize'));
					$('#main-product-img-lightbox-anchor-@Model.Id').attr('title', $(this).attr('title'));
					$('.picture-caption').text($(this).attr('title'));
                });
            });
		</script>
	}
</div>
```

**Comment by DmitriyKulagin** [comment]
@cklenk looks like good idea. please make a pull request

**Comment by DmitriyKulagin** [comment]
Closed #3405
